docs/content/docs/data-structures/map.md

```diff
@@ -1,10 +1,449 @@
+            "email": "joeschmo@email.com",
+            "address": "1234 Fluid Way"
+        }
+    },
+    "task2": {
+        "title": "Even More Awesome Task",
+        "description": "Doing even more awesome things",
+        "assignedTo": {
+            "name": "Jane Doe",
+            "email": "janedoe@email.com",
+            "address": "5678 Framework Street"
+        }
+    }
+}
+```
+Here, you can have each task itself be its own `SharedMap` and have a parent `SharedMap` hold all of the handles to each task.
+
+Then, the initial map would look like:
+```json
+{
+    "task1": task1MapHandle,
+    "task2": task2MapHandle
+}
+```
+And the `task1` map would have:
+```json
+{
+    "title": "Awesome Task",
+    "description": "Doing the most awesome things",
+    "assignedTo": {
+        "name": "Joe Schmo",
+        "email": "joeschmo@email.com",
+        "address": "1234 Fluid Way"
+    }
+}
+```
+And the `task2` map would have:
+```json
+{
+    "title": "Even More Awesome Task",
+    "description": "Doing even more awesome things",
+    "assignedTo": {
+        "name": "Jane Doe",
+        "email": "janedoe@email.com",
+        "address": "5678 Framework Street"
+    }
+}
+```
+With this nested map structure, you are able to both ensure that each field that will be collaboratively edited is stored separately and that you have a hierarchy in how you store the data in your `SharedMap` that reflects the app's data model.
+
+Whenever a new task is created, you can call `container.create` to get a new `SharedMap` instance and store its handle into the `initialMap` that is provided as an `initialObject`. Since each additional task map would need its own unique key, you can use a random string generator, such as [uuid](https://www.npmjs.com/package/uuid), to create a new one each time and avoid any clashes.
+
+```javascript
+const schema = {
+    name: "example-container",
+    initialObjects: {
+        initalMap: SharedMap,
+    },
+    dynamicObjectTypes: [SharedMap]
+}
+
+const { fluidContainer, containerServices } = await client.getContainer(/*service config*/, schema);
+
+const initialMap = fluidContainer.initialObjects.initialMap;
+
+const newTask = fluidContainer.create(SharedMap);
+const newTaskId = uuid();
+
+initialMap.set(newTaskId, newTask.handle);
+```
+
+At this point, you can use `intialMap.keys()` to see all of the various task IDs that are available or `initialMap.values()` to return all of the task handles.
+
+Say you wanted to fetch back a specific task, with ID `task123` and allow the user to edit its description.
+
+```javascript
+const taskHandle = initialMap.get("task123");
+const task = await taskHandle.get();
+task.set("description", editedDescription)
+```
+
+Since each task is being in a separate map and all of the fields within the task object are being stored in their own unique keys, you can now have both a hiearchical structure in your data and avoid any clashes in edits between multiple users.
```

Comment:
```suggestion
Since each task is stored in a separate SharedMap and all of the fields within the task object are being stored in their own unique keys, your data now has a hierarchical structure that reflects the app's data model while individual tasks' properties can be edited independently.

```